vault/resource_pki_secret_backend_sign.go

```diff
@@ -194,16 +206,46 @@ func pkiSecretBackendSignCreate(d *schema.ResourceData, meta interface{}) error
 	d.Set("issuing_ca", resp.Data["issuing_ca"])
 	d.Set("ca_chain", resp.Data["ca_chain"])
 	d.Set("serial", resp.Data["serial"])
+	d.Set("expiration", resp.Data["expiration"])
 
 	d.SetId(fmt.Sprintf("%s/%s/%s", backend, name, commonName))
 	return pkiSecretBackendSignRead(d, meta)
 }
 
+func pkiSecretBackendSignDiff(d *schema.ResourceDiff, meta interface{}) error {
+	if d.Id() == "" {
+		return nil
+	}
+
+	expiration := d.Get("expiration").(int)
+	expireTime := time.Unix(int64(expiration), 0)
+
+	minSeconds := 0
+	if v, ok := d.GetOk("min_seconds_remaining"); ok {
+		minSeconds = v.(int)
+	}
+
+	renewTime := expireTime.Add(-time.Duration(minSeconds) * time.Second)
+	if time.Now().After(renewTime) {
+		log.Printf("[DEBUG] certificate %q is due for renewal, expires %s, renewal time %s", d.Id(), expireTime, renewTime)
+		d.SetNewComputed("certificate")
+		return nil
+	}
+
+	log.Printf("[DEBUG] certificate %q is not due for renewal, expires %s, renewal time %s", d.Id(), expireTime, renewTime)
+	return nil
```

Comment:
It would be great to get test coverage on all the code changes in this file too - the new fields that have been added, and the certificate that's generated.